cassandra/cluster.py

```diff
@@ -4240,6 +4239,8 @@ def wait_for_schema_agreement(self, connection=None, preloaded_results=None, wai
             select_peers_query = self._get_peers_query(self.PeersQueryType.PEERS_SCHEMA, connection)
 
             while elapsed < total_timeout:
+                connection = target_connection or self._connection
+
                 peers_query = QueryMessage(query=maybe_add_timeout_to_query(select_peers_query, self._metadata_request_timeout),
                                            consistency_level=cl)
```

Comment:
I don't see how this fixes the problem, and it is not explained in the commit message or PR description.
In the linked issue `wait_for_response` throws `ConnectionShutdown` exception, which as far as I can tell will be re-thrown - so the function will fail even with this change.

Reply:
Thanks, true, take a look at https://github.com/scylladb/python-driver/pull/461